src/unit_test.c

```diff
@@ -0,0 +1,340 @@
+/***********************************************************************
+ * Copyright (c) 2025  Matias Furszyfer (furszy)                       *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#if SUPPORTS_CONCURRENCY
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#endif
+
+#include "unit_test.h"
+#include "testrand.h"
+#include "tests_common.h"
+
+#define UNUSED(x) (void)(x)
+
+/* Number of times certain tests will run */
+int COUNT = 16;
+
+static int parse_jobs_count(const char* key, const char* value, struct TestFramework* tf);
+static int parse_iterations(const char* key, const char* value, struct TestFramework* tf);
+static int parse_seed(const char* key, const char* value, struct TestFramework* tf);
+
+/* Mapping table: key → handler */
```

Comment:
meta-nit: seeing this fancy single-character arrow made me wonder if this file is UTF-8 encoded, and according to the `file` utility it indeed is:
```
$ file ./src/unit_test.c
./src/unit_test.c: C source, Unicode text, UTF-8 text
```
Seems that other source files in the repo are not (see `$ file ./src/*.h ./src/*.c`, resulting all in "`C source, ASCII text`"). I could imagine that in theory really old compilers could struggle with that and using ASCII encoding would be the safe choice, but not sure if that concern is practically relevant at all.